lib/api/controller/document.js

```diff
@@ -221,40 +225,35 @@ class DocumentController extends NativeController {
    * @param {Request} request
    * @returns {Promise<Object>}
    */
-  createOrReplace (request) {
-    let modifiedRequest;
-    let response;
-
+  async createOrReplace (request) {
     const id = this.getId(request);
     const content = this.getBody(request);
     const userId = this.getUserId(request);
     const refresh = this.getString(request, 'refresh', 'false');
     const { index, collection } = this.getIndexAndCollection(request);
 
-    return this.kuzzle.validation.validate(request, false)
-      .then(newRequest => {
-        modifiedRequest = newRequest;
-
-        return this.publicStorage.createOrReplace(
-          index,
-          collection,
-          id,
-          content,
-          { refresh, userId });
-      })
-      .then(_response => {
-        // response: { _id, _version, _source, created }
-        response = _response;
-
-        if (response.created) {
-          return this.kuzzle.notifier.notifyDocumentCreate(
-            modifiedRequest,
-            response);
-        }
-
-        return this.kuzzle.notifier.notifyDocumentReplace(modifiedRequest);
-      })
-      .then(() => response);
+    const modifiedRequest = await this.kuzzle.validation.validate(request, false);
+
+    // response: { _id, _version, _source, created }
```

Comment:
I don't get the purpose of this comment

Reply:
That's an old one. Probably a reminder of what's returned by the ES request :thinking: 
Should I remove it?

Reply:
I don't think so but current format is not very useful IMHO 🙂